django/utils/decorators.py

```diff
@@ -152,6 +152,18 @@ def callback(response):
     return _make_decorator
 
 
+def async_middleware(func):
+    """
+    Mark a middleware factory as returning an async middleware.
+    """
+    @wraps(func)
+    def wrapper(request):
+        return func(request)
```

Comment:
Do you need to do this? I think you can just have this be `func._is_async = True; return func`, no?